src/codeview.js

```diff
@@ -556,10 +572,35 @@ var Codeview = GObject.registerClass({
         }
     }
 
+    dbusRegister(appId, id) {
+        const objPath = Gio.Application.get_default().get_dbus_object_path();
+        const winName = appId.replace(/\./gi, '_');
+        this.objectPath = `${objPath}/window/${winName}/codeview/${id}`;
+        this.address = `com.hack_computer.HackToolbox.window.${winName}.codeview.${id}`;
+        this._dbus = Gio.DBusExportedObject.wrapJSObject(CodeViewIface, this);
+        try {
+            this._dbus.export(Gio.DBus.session, this.objectPath);
+        } catch (e) {
+            logError(e);
+        }
+    }
+
+    dbusUnregister() {
+        if (this._dbus)
+            this._dbus.unexport();
+    }
+
     setCompileResults(results, userInitiated = true) {
         const [begin, bound] = this._buffer.get_bounds();
         this._buffer.remove_source_marks(begin, bound, MarkType.ERROR);
         this._buffer.remove_tag(this._errorStyle, begin, bound);
+
+        if (this._dbus) {
+            this._errors = results.length;
```

Comment:
If we have `errors` as a GObject property, we can just call `emit_property_changed` on the setter or connecting to `notify::errors` and do it on its handler.